src/main/java/cws/k8s/scheduler/memory/LinearPredictor.java

```diff
@@ -0,0 +1,108 @@
+ * this work. If not, see <https://www.gnu.org/licenses/>. 
+ */
+
+package cws.k8s.scheduler.memory;
+
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.apache.commons.math3.stat.regression.SimpleRegression;
+
+import cws.k8s.scheduler.model.Task;
+import lombok.extern.slf4j.Slf4j;
+
+//@formatter:off
+/**
+ * LinearPredictor will use the following strategy:
+ * 
+ * If there are less than 2 observations, give no prediction, else:
+ * Calculate linear regression model and provide predictions.
+ * 
+ * Predictions start with 10% over-provisioning. If tasks fail, this will
+ * increase automatically.
+ * 
+ * @author Florian Friederici
+ *
+ */
+//@formatter:on
+@Slf4j
+public class LinearPredictor implements MemoryPredictor {
+    
+    Map<String, SimpleRegression> model;
+    Map<String, Double> overprovisioning;
+    
+    public LinearPredictor() {
+        model = new HashMap<>();
+        overprovisioning = new HashMap<>();
+    }
+    
+    @Override
+    public void addObservation(Observation o) {
+        log.debug("LinearPredictor.addObservation({})", o);
+        if (!TaskScaler.checkObservationSanity(o)) {
+            log.warn("dismiss observation {}", o);
+            return;
+        }
+
+        if (!overprovisioning.containsKey(o.task)) {
+            overprovisioning.put(o.task, 1.1);
+        }
+        
+        if (Boolean.TRUE.equals(o.success)) {
+            if (!model.containsKey(o.task)) {
+                model.put(o.task, new SimpleRegression());
+            }
+            
+            double x = o.getInputSize();
+            double y = o.getPeakRss().doubleValue();
+            model.get(o.task).addData(x,y);
+        } else {
+            log.debug("overprovisioning value will increase due to task failure");
+            Double old = overprovisioning.get(o.task);
+            overprovisioning.put(o.task, old+0.05);
+        }
+    }
+
+    @Override
+    public String queryPrediction(Task task) {
+        String taskName = task.getConfig().getTask();
+        log.debug("LinearPredictor.queryPrediction({},{})", taskName, task.getInputSize());
+
+        if (!model.containsKey(taskName)) {
+            log.debug("LinearPredictor has no model for {}", taskName);
+            return null;
+        }
+        
+        SimpleRegression simpleRegression = model.get(taskName);
+        double prediction = simpleRegression.predict(task.getInputSize());
+
+        if (Double.isNaN(prediction)) {
+            log.debug("No prediction possible for {}", taskName);
+            return null;
+        }
+
+        if (prediction < 0) {
+            log.warn("prediction would be negative: {}", prediction);
+            return null;
+        }
+
+        return BigDecimal.valueOf(prediction).multiply(BigDecimal.valueOf(overprovisioning.get(taskName))).setScale(0, RoundingMode.CEILING).toPlainString();
```

Comment:
Ok here are two different issues:

1) The interface
`public String queryPrediction(Task task)`
This is currently a String, because the patchTask requires a String.

2) Low predictions
Are currently catched in TaskScaler:214